### ISSUE TITLE
[2.2.9] @entangle not working with nested array (copy pastable!) 

### ISSUE THREAD

**Issue by basepack** [livewire/livewire#1762]
### Description
Yes this is related to https://github.com/livewire/livewire/issues/1682 and https://github.com/livewire/livewire/discussions/1750 and it should be fixed in 19df71a, but it's not.

I am experiencing almost the same as the guys above.

**@calebporzio Yes! Contains Copy Pastable Code** - as per podcast that is what you like :-)

Gif for some context:
![example](https://user-images.githubusercontent.com/939500/95201248-f1288c80-07df-11eb-9c0b-e94f99a6835c.gif)

### Stripped-down, copy-pastable code snippets
```php
class Openinghours extends Component
{
    public ?array $openinghours;

    public function mount()
    {
        $this->openinghours = [
            'monday'    => ['from' => '09:00',],
            'tuesday'   => ['from' => '09:00',],
            'wednesday' => ['from' => '09:00',],
        ];
    }

    public function ddIt()
    {
        dd($this->openinghours);
    }
}
```

```blade
<div x-data="{ openinghours: @entangle('openinghours') }">
    <div class="flex">
        <div
            x-on:click="openinghours.monday.from = '12:00'"
            dusk="button"
            class="p-4 bg-blue-300">
            monday

            <div>output Alpine:
                <span dusk="output.alpine" x-text="openinghours.monday.from"></span>
            </div>

            <div>output $wire:
                <span dusk="output.wire" x-text="$wire.openinghours.monday.from"></span>
            </div>

            <div>output blade:
                <span dusk="output.blade">{{ $openinghours['monday']['from'] }}</span>
            </div>

        </div>
        <div wire:click="ddIt" class="p-4 bg-orange-300 cursor-pointer">dd $openinghours from Livewire</div>
    </div>
</div>

```

Dusk test (I don't have Dusk working yet, but it should work)
```php
$this->browse(function ($browser) {
    Livewire::visit($browser, Openinghours::class)
        ->assertSeeIn('@output.alpine', '09:00')
        ->assertSeeIn('@output.wire', '09:00')
        ->assertSeeIn('@output.blade', '09:00')
        ->waitForLivewire()->click('@button')
        ->assertSeeIn('@output.alpine', '12:00')
        ->assertSeeIn('@output.wire', '12:00') // will fail
        ->assertSeeIn('@output.blade', '12:00'); // will fail
});
```

### Exact steps to reproduce
- create component with array
- in mount method fill that array
- in view change a nested.array.property = newvalue
- see that newvalue is only update in the Alpine side and not in $wire or blade

**Context**
- Livewire version: 2.2.9
- Laravel version: 8.8.0
- Browser: Safari.. just kidding it's Chrome

**Comment by uteq** [comment]
I think this is related https://github.com/livewire/livewire/issues/1753
The Jetstream component also uses @entangle, and after I upgraded to Livewire 2.2.9 using an array stoped working.

**Comment by calebporzio** [comment]
Thanks for the well written/reproducible issue @basepack.

Unfortunately, this is a limitation of Alpine's reactivity system. Alpine has no way currently of detecting deep changes on Arrays, so in this case, Livewire never knows to update it's own property.

You can always force reactivity by overwriting the array completely (instead of setting a deep property only) like so:
```html
x-on:click="let foo = {...openinghours}; foo.monday.from = '12:00'; openinghours = foo"
```

Supporting deep observability in Alpine is something that has been talked about and is on the wish list.

Thanks!

**Comment by basepack** [comment]
Well thanks very much @calebporzio I didn't think of Alpine not being able to do that.

Also thanks for your proposed solution, however after a full day of trial and error yesterday I came up with the following easy solution.

I flattened my opening hour array to:
```php
        $this->openinghours = [
            ['monday', 'from' => '09:00',],
            ['monday', 'from' => '09:00',],
            ['wednesday', 'from' => '09:00',],
        ];
```

Then in blade:
```blade
@foreach($this->openinghours as $key => $openingHour)
    <input 
        wire:model.defer="openinghours.{{ $key }}.from" 
        type="time"/>
@endforeach
```

I cannot believe I deleted around 250 lines of code consisting of Alpine x-models/$dispatch/x-on:time-changed, Livewire events/update functions and what not with just that single `wire:model.defer="openinghours.{{ $key }}.from" ` blade line above.

I leave this here for future reference for anyone who need this.
